engine/src/test/java/org/hibernate/search/test/searchfactory/SearchFactoryTest.java

```diff
@@ -49,6 +67,158 @@ public void testTypeWithNoDocumentIdThrowsException() {
 	}
 
 	@Test
+	@Category(ElasticsearchSupportInProgress.class) // HSEARCH-2481 Byteman-based tests re-executed in the Elasticsearch module won't work
+	@TestForIssue(jiraKey = "HSEARCH-2277")
+	@BMRules(rules = {
+			@BMRule(
+					name = "Skip failure on first index manager initialization",
+					isInterface = true,
+					targetClass = "org.hibernate.search.indexes.spi.IndexManager",
+					targetMethod = "initialize(String, Properties, Similarity, WorkerBuildContext)",
+					helper = "org.hibernate.search.testsupport.BytemanHelper",
+					condition = "NOT flagged(\"mustFailIndexManagerInitialization\")",
+					action = "flag(\"mustFailIndexManagerInitialization\")"
+			),
+			@BMRule(
+					name = "Simulate failure on second index manager initialization",
+					isInterface = true,
+					targetClass = "org.hibernate.search.indexes.spi.IndexManager",
+					targetMethod = "initialize(String, Properties, Similarity, WorkerBuildContext)",
+					helper = "org.hibernate.search.testsupport.BytemanHelper",
+					condition = "flagged(\"mustFailIndexManagerInitialization\")",
+					action = "simulateFailure()"
+			),
+			@BMRule(
+					name = "Track calls to ServiceManager.releaseAllServices",
+					targetClass = "org.hibernate.search.engine.service.impl.StandardServiceManager",
+					targetMethod = "releaseAllServices()",
+					helper = "org.hibernate.search.testsupport.BytemanHelper",
+					action = "countInvocation()"
+			)
+	} )
+	public void testServicesStoppedAfterIndexInitializationException() {
+		SearchConfigurationForTest cfg = new SearchConfigurationForTest()
+				.addClass( AnnotatedClass.class ).addClass( SecondAnnotatedClass.class );
+
+		try {
+			new SearchIntegratorBuilder().configuration( cfg ).buildSearchIntegrator();
+		}
+		catch (SearchException e) {
+			// Ignore the exception
```

Comment:
I would actually do something here so that the test can assert that the `SearchException` was thrown.

e.g. flip a boolean, so that you can assert on the boolean after the catch block?

Otherwise you have the risk that the Services are indeed released, but just because the failures didn't happen.

I'm being extra paranoid here because the Byteman rules tend to break without us noticing when they rely on specific signatures...